pkg/adsc/adsc.go

```diff
@@ -153,6 +153,9 @@ type ADSC struct {
 	// restarts.
 	LocalCacheDir string
 
+	// Synchronization for receiving the gRPC stream.
+	RecvWg sync.WaitGroup
```

Comment:
Private ?

Reply:
Also the comment is not very clear - what is it syncing ? From the code - active connections ?

Reply:
The gRPC stream for ADS is handled in a goroutine, and there is not a way of telling when the goroutine has finished. When adding the tests, I found this to be a problem because the test would exit without finishing the goroutine & finishing handling the stream. So, this Wg will automatically start with a delta of 1 and third parties can wait on it if they need to do something synchronously when the goroutine finished. Does that make sense?

Reply:
I meant: the comment should say this. "sync for receiving gRPC stream" is pretty generic. 

What you put in your reply ( maybe a shorter version :-) should be in the code comment.

Reply:
sure, ok.